saleor/static/dashboard-next/siteSettings/views/index.tsx

```diff
@@ -0,0 +1,174 @@
+import * as React from "react";
+
+import Messages from "../../components/messages";
+import Toggle from "../../components/Toggle";
+import i18n from "../../i18n";
+import { maybe } from "../../misc";
+import { AuthorizationKeyType } from "../../types/globalTypes";
+import SiteSettingsKeyDialog, {
+  SiteSettingsKeyDialogForm
+} from "../components/SiteSettingsKeyDialog";
+import SiteSettingsPage, {
+  SiteSettingsPageFormData
+} from "../components/SiteSettingsPage";
+import {
+  TypedAuthorizationKeyAdd,
+  TypedAuthorizationKeyDelete,
+  TypedShopSettingsUpdate
+} from "../mutations";
+import { TypedSiteSettingsQuery } from "../queries";
+import { AuthorizationKeyAdd } from "../types/AuthorizationKeyAdd";
+import { AuthorizationKeyDelete } from "../types/AuthorizationKeyDelete";
+import { ShopSettingsUpdate } from "../types/ShopSettingsUpdate";
+
+export const SiteSettings: React.StatelessComponent<{}> = () => (
+  <Messages>
+    {pushMessage => (
+      <Toggle>
+        {(openedAddKeyDialog, { toggle: toggleAddKeyDialog }) => {
+          const handleAddKeySuccess = (data: AuthorizationKeyAdd) => {
+            if (!maybe(() => data.authorizationKeyAdd.errors.length)) {
+              pushMessage({
+                text: i18n.t("Authorization key added", {
+                  context: "notification"
+                })
+              });
+              toggleAddKeyDialog();
+            }
+          };
+          const handleDeleteKeySuccess = (data: AuthorizationKeyDelete) => {
+            if (!maybe(() => data.authorizationKeyDelete.errors.length)) {
+              pushMessage({
+                text: i18n.t("Authorization key deleted", {
+                  context: "notification"
+                })
+              });
+            } else {
+              pushMessage({
+                text: i18n.t(
+                  "Could not delete authorization key: {{ message }}",
+                  {
+                    context: "notification",
+                    message: data.authorizationKeyDelete.errors[0].message
+                  }
+                )
+              });
+            }
+          };
+          const handleSiteSettingsSuccess = (data: ShopSettingsUpdate) => {
+            if (
+              !maybe(() => data.shopDomainUpdate.errors.length) &&
```

Comment:
Using `maybe` in a conditional can lead to unexpected results.

Reply:
There are only 3 possibilities here:
- `errors === null` => `false`
- `errors.length === 0` => `false`
- `errors.length > 0` => `true`

So it's not like there are N possible results and we can't list them.